inspirehep/modules/workflows/tasks/arxiv.py

```diff
@@ -215,68 +212,116 @@ def arxiv_plot_extract(obj, eng):
+            obj.data['authors'] = extracted_authors
+
+            set_mark(obj, "authors_xml", True)
+        else:
+            set_mark(obj, "authors_xml", False)
+
+
+def extract_authors_from_xml(xml_content):
+    builder = LiteratureBuilder()
+    if isinstance(xml_content, str):
+        xml_content = xml_content.decode('utf-8')
+
+    # Probably the %auto-ignore comment exists, so we skip the
+    # first line. See: inspirehep/inspire-next/issues/2195
+    if "%auto-ignore" in xml_content:
+        xml_content = xml_content.split('\n', 1)[1]
+
+    content = Selector(text=xml_content, type="xml")
+
+    content.remove_namespaces()
+    # Going through all the authors in the file
+    for author in content.xpath("//Person"):
+
+        ids = []
+        affiliations = []
+        affiliations_identifiers = []
+
+        # Getting all the author ids
+        for source, id in itertools.izip(author.xpath('./authorIDs/authorID[@source!="" and text()!=""]/@source | ./authorids/authorid[@source!="" and text()!=""]/@source').getall(), author.xpath('./authorIDs/authorID[@source!="" and text()!=""]/text() | ./authorids/authorid[@source!="" and text()!=""]/text()').getall()):
+            if not re.match("undefined", source, flags=re.IGNORECASE) or not re.match("undefined|inspire-", id, flags=re.IGNORECASE):
+                if source == u'CCID':
+                    ids.append(['CERN', id])
+                elif source == u'INSPIRE':
+                    ids.append(['{} ID'.format(source), id])
+                else:
+                    ids.append([source, id])
+
+        # Getting all the names for affiliated organizations using the organization ids from author
+        for affiliation in author.xpath("./authorAffiliations/authorAffiliation/@organizationid").getall():
+            affiliations.append(content.xpath('//organizations/Organization[@id="{}"]/orgName[@source="spiresICN" or @source="INSPIRE" and text()!="undefined" and text()!="UNDEFINED" and text()!="" ]/text()'.format(affiliation)).get())
+
+            # Getting all the affiliations_identifiers for affiliated organizations using the organization ids from author
+
+            for value, source in itertools.izip(content.xpath('//organizations/Organization[@id="{}"]/orgName[@source="ROR" or @source="GRID" and text()!=""]/text()'.format(affiliation)).getall(), content.xpath('//organizations/Organization[@id="{}"]/orgName[@source="ROR" or @source="GRID" and text()!=""]/@source'.format(affiliation)).getall()):
+                if not re.match("undefined", source, flags=re.IGNORECASE) or not re.match("undefined", id, flags=re.IGNORECASE):
+
+                    if source == 'ROR' and not re.match("https://ror.org/*", value):
+                        value = 'https://ror.org/{}'.format(value)
+
+                    affiliations_identifiers.append([source, value])
+
+        name = normalize_name(author.xpath('.//familyName/text()').get() + ', ' + author.xpath('.//givenName/text()').get())
```

Comment:
```suggestion
        name = normalize_name("{}, {}".format(author.xpath('.//familyName/text()').get(), author.xpath('.//givenName/text()').get()))
```

Reply:
sorry, python 2, so `u"{}, {}"`

Reply:
This is why one should never copy their homework from others ;)